source/_patterns/01-elements/input/input.scss

```diff
@@ -1,50 +1,159 @@
+@use "@db-ui/base/build/scss/icon/icons.helpers" as *;
+
+$icons-path: "../../../icons/" !default;
+
+@mixin label-focus-animation(
+	$translationX: var(--elm-input-padding-inline-start)
+) {
+	$transformScale: -1.25;
+
+	transition: opacity 0.08s, transform 0.15s;
+	opacity: var(--db-input-focus-opacity);
+
+	transform: translate($translationX, calc($db-sizing-md * $transformScale))
+		scale(var(--db-input-label-scale));
+}
 
 .elm-input {
-	@extend %form-element;
+	--elm-input-padding-inline-start: #{$db-spacing-fixed-sm};
+	--elm-input-padding-inline-end: #{$db-spacing-fixed-sm};
 
-	color: $input---color;
+	position: relative;
 
-	font-size: to-rem($pxValue: 16);
-	height: to-rem($pxValue: 44);
-	line-height: to-rem($pxValue: 24);
-	max-width: 100%;
+	// padding-inline-end according to current amount of icons
+	&[data-icon-before] {
+		--elm-input-padding-inline-start: #{calc(
+				2 * $db-spacing-fixed-sm + 0.425rem
+			)};
+
+		input {
+			padding-inline-start: var(--elm-input-padding-inline-start);
+		}
+	}
+
+	&[data-icon-after],
+	&[data-variant] {
+		--elm-input-padding-inline-end: #{calc(
+				2 * #{$db-spacing-fixed-sm} + 0.425rem
+			)};
+
+		input {
+			padding-inline-end: var(--elm-input-padding-inline-end);
+		}
+	}
 
-	padding: $db-spacing-fixed-regular-xs $db-spacing-fixed-regular-m;
+	&[data-variant][data-icon-after] {
+		--elm-input-padding-inline-end: #{calc(
+				2 * #{$db-spacing-fixed-sm} + 0.425rem + 0.75rem + #{$db-spacing-fixed-xs}
+			)};
 
+		input {
+			padding-inline-end: var(--elm-input-padding-inline-end);
+		}
+
+		.elm-state-icon::after {
+			margin-inline-end: calc(0.75rem + $db-spacing-fixed-xs);
+		}
+	}
+
+	&::before,
+	&::after {
+		--icon-font-family: var(--db-base-icon-font-family);
+		--icon-font-size: var(--db-base-icon-font-size);
+
+		@extend %db-body-sm;
+
+		height: var(--db-sizing-md);
+		display: flex;
+		align-items: center;
+		position: absolute;
+		top: 0;
+		padding: 0 $db-spacing-fixed-xs;
+	}
+
+	&::after {
+		right: 0;
+	}
+
+	// currently not supported by firefox
+	&:has(> input:disabled) {
+		&::before,
+		&::after,
+		.elm-state-icon::after {
+			opacity: 0.5;
+		}
+	}
+}
+
+.elm-input input {
+	@extend %db-bg-neutral-transparent-semi;
+
+	border: 1px solid transparent;
+	border-radius: 4px;
+	height: $db-sizing-md;
+	max-width: 100%;
 	width: 100%;
+	padding: $db-spacing-fixed-xs var(--elm-input-padding-inline-end)
+		$db-spacing-fixed-xs var(--elm-input-padding-inline-start);
 
-	&:not(:placeholder-shown) {
-		padding-bottom: 0;
-		padding-top: to-rem($pxValue: 16);
+	& + label {
+		position: absolute;
+		display: flex;
+		align-items: center;
+		font-size: var(--db-type-body-font-size-md);
+		color: $db-colors-secondary-enabled;
+
+		transform-origin: left;
+		transform: translate(
+			var(--elm-input-padding-inline-start),
+			calc(-1 * var(--db-sizing-md))
+		);
+		transition: transform 0.2s $db-transition-functional-timing;
+
+		height: $db-sizing-md;
+		pointer-events: none;
```

Comment:
this declaration prevents the clicking / pointing on the `label` and `focus`ing the input element afterwards (even after the label floated away). Do we really need this declaration ?